frame/election-provider-support/src/lib.rs

```diff
@@ -662,16 +658,235 @@ impl<AccountId: IdentifierT, Accuracy: PerThing128, Balancing: Get<Option<Balanc
+		self.0 == 0
+	}
+	fn zero() -> Self {
+		CountBound(0)
+	}
+}
+
+/// Size bound of data provider bounds.
+#[derive(Clone, Copy, Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub struct SizeBound(pub u32);
+
+impl Zero for SizeBound {
+	fn is_zero(&self) -> bool {
+		self.0 == 0
+	}
+	fn zero() -> Self {
+		SizeBound(0)
+	}
+}
+
+impl Add for SizeBound {
+	type Output = Self;
+	fn add(self, rhs: Self) -> Self::Output {
+		SizeBound(self.0 + rhs.0)
+	}
+}
+
+impl From<u32> for SizeBound {
+	fn from(value: u32) -> Self {
+		SizeBound(value)
+	}
+}
+
 sp_core::generate_feature_enabled_macro!(runtime_benchmarks_enabled, feature = "runtime-benchmarks", $);
 sp_core::generate_feature_enabled_macro!(runtime_benchmarks_or_fuzz_enabled, any(feature = "runtime-benchmarks", feature = "fuzzing"), $);
+
+/// Data provider limits that can be bounded based on the count of elements or the scale encoded
+/// size of the final result in MB. It can be used to represent the bounds of election targets
+/// and voters or any other future unit.
+#[derive(Clone, Copy, Default, Debug)]
+pub struct DataProviderBounds {
+	pub count: Option<CountBound>,
+	pub size: Option<SizeBound>,
+}
+
+impl DataProviderBounds {
+	/// Unbonded data provider limit.
+	pub fn new_unbounded() -> Self {
+		DataProviderBounds { count: None, size: None }
+	}
+
+	///  Returns true if `given_count` exhausts `self.count`.
+	pub fn count_exhausted(self, given_count: CountBound) -> bool {
+		self.count.map_or(false, |count| given_count > count)
+	}
+
+	///  Returns true if `given_size` exhausts `self.size`.
+	pub fn size_exhausted(self, given_size: SizeBound) -> bool {
+		self.size.map_or(false, |size| given_size > size)
+	}
+
+	/// Returns true if `given_size` or `given_count` exhausts `self.size` or `self_count`,
+	/// respectively.
+	pub fn exhausted(self, given_size: Option<SizeBound>, given_count: Option<CountBound>) -> bool {
+		self.count_exhausted(given_count.unwrap_or(CountBound::zero())) ||
+			self.size_exhausted(given_size.unwrap_or(SizeBound::zero()))
+	}
+
+	/// Returns an instance of `Self` that is constructed by capping both the `count` and `size`
+	/// fields. If `self` is None, overwrite it with the provided bounds.
+	pub fn max(self, bounds: DataProviderBounds) -> Self {
+		DataProviderBounds {
+			count: self
+				.count
+				.map(|c| {
+					c.clamp(CountBound::zero(), bounds.count.unwrap_or(CountBound(u32::MAX))).into()
+				})
+				.or(bounds.count),
+			size: self
+				.size
+				.map(|c| {
+					c.clamp(SizeBound::zero(), bounds.size.unwrap_or(SizeBound(u32::MAX))).into()
+				})
+				.or(bounds.size),
+		}
+	}
+}
+
+/// The limits of an election result. The bounds are defined over the count of element of the
+/// election (voters or targets) or the overall size of the elements in MB.
+#[derive(Clone, Debug)]
+pub struct ElectionBounds {
+	pub voters: DataProviderBounds,
+	pub targets: DataProviderBounds,
+}
+
+/// Utility builder for [`ElectionBounds`].
```

Comment:
```suggestion
/// Utility builder for [`ElectionBounds`].
///
/// ## Examples
///
/// ```
/// use frame_election_provider_support::ElectionBoundsBuilder;
///
/// # fn main() {
///
/// # }
/// ```
```
In the spirit of an up the docs game? I think compiling test in rust-doc is always a great idea. You can cover a few example of how the builder can be used to build common example, or edge-cases, if any.

Reply:
Basically I would move some or all of `mod elections_bounds` into rust-docs.